### ISSUE TITLE
DDPG console log prints re-derived post-update loss_value that diverges from stored value

### ISSUE THREAD

**Issue by kwojdalski** [kwojdalski/masters_thesis#294]
**File:** src/trading_rl/trainers/ddpg.py:164-222; src/trading_rl/trainers/base.py:525-533
**Category:** 2 Wrong Variable Used
**Severity:** LOW

**What it computes:**
```python
# ddpg.py:164-222
loss_vals = self.ddpg_loss(sample)              # FIRST forward pass
...
loss_vals["loss_value"].backward()               # the value actually optimized
...
self.ddpg_loss.value_network_params.to_module(self.value_net)   # critic weights updated

loss_vals_actor = self.ddpg_loss(sample)         # SECOND forward pass, post-critic-update
...
actor_loss = loss_vals_actor["loss_actor"].item()   # correct (fresh) actor loss
value_loss = loss_vals["loss_value"].item()         # correct (pre-update) value loss
self.logs["loss_value"].append(value_loss)          # stores the PRE-update value
self.logs["loss_actor"].append(actor_loss)
...
self._log_progress(max_length, buffer_len, loss_vals_actor)   # only passes loss_vals_actor
```
```python
# base.py:527
curr_loss_value = loss_vals[self._value_loss_key].item()
```
`_log_progress` is only ever passed `loss_vals_actor` (the second, post-critic-update forward pass), so `curr_loss_value` reads `loss_vals_actor["loss_value"]` — a value loss re-derived on the critic's *already-updated* parameters against the same batch it was just trained on (typically lower than the pre-update loss that was actually backpropagated). `DDPGLoss.forward()` always returns both `loss_actor` and `loss_value` in one call, so this doesn't crash — it silently returns a mismatched number. This diverges from `self.logs["loss_value"]` (ddpg.py:208-209, correctly `loss_vals["loss_value"]`, pre-update) and from the `value_loss` sent to `callback.log_training_step` (ddpg.py:218).

This is the mirror-image of the fix applied for issue #261 (which made the printed `loss_actor` consistent with `self.logs["loss_actor"]` by switching `_log_progress`'s input to `loss_vals_actor`) — that fix made `loss_actor` correct but left `loss_value` reading from the wrong dict. Only the human-readable console log is affected; persisted metrics (`self.logs`, MLflow) are correct.

**What it should compute:** The console log's `loss_value` should match the pre-update value that is already stored in `self.logs["loss_value"]` (i.e. `loss_vals["loss_value"]`, not `loss_vals_actor["loss_value"]`).

**Proposed fix:** Override `_log_progress` in `DDPGTrainer` (or extend the base method to accept separate value/actor loss dicts) to read `loss_value` from `loss_vals` and `loss_actor` from `loss_vals_actor`, mirroring how `SACTrainer._log_progress` already handles three separate loss dicts (sac.py:245-263).
